source/extensions/filters/network/common/redis/fault_impl.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.pb.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/common/redis/fault.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class Fault {
+public:
+  Fault(envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+  FaultType faultType() const { return fault_type_; };
+  std::chrono::milliseconds delayMs() const { return delay_ms_; };
+  const std::vector<std::string> commands() const { return commands_; };
+  uint64_t defaultValue() const { return default_value_; };
+  absl::optional<std::string> runtimeKey() const { return runtime_key_; };
+
+private:
+  static std::vector<std::string> buildCommands(
+      envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+  FaultType fault_type_;
+  std::chrono::milliseconds delay_ms_;
+  const std::vector<std::string> commands_;
+  uint64_t default_value_;
+  absl::optional<std::string> runtime_key_;
+};
+
+using FaultPtr = std::shared_ptr<Fault>;
```

Comment:
It seems both Fault and the FaultMap are immutable post construction.  Can we make them const?

Reply:
Done.